### PR TITLE
Better error message when an image reference fails to parse

### PR DESCRIPTION
**Pull request by ekcasey**
* Fixes: #295

Doesn't validate that all destination tags have a matching registry when
exporting to the daemon (this is unnecessary in the daemon case)

### CODE REVIEW

File: image/image_test.go [hunk 1]
```diff
@@ -14,19 +14,34 @@ func TestImage(t *testing.T) {
...
 		when("a single registry is provided", func() {
 			it("does not return an error", func() {
-				err := image.EnsureSingleRegistry("gcr.io/some/repo", "gcr.io/other-repo:latest", "gcr.io/final-repo")
+				err := image.ValidateDestinationTags(false, "gcr.io/some/repo", "gcr.io/other-repo:latest", "gcr.io/final-repo")
 				h.AssertNil(t, err)
 			})
 		})
+
+		when("the tag reference is invalid", func() {
+			it("error", func() {
```
Comment (natalieparellano) [inline image/image_test.go:41]:
```suggestion
			it("errors", func() {
```

File: cmd/lifecycle/creator.go [hunk 2]
```diff
@@ -77,8 +77,8 @@ func (c *createCmd) Args(nargs int, args []string) error {
 		c.previousImage = c.imageName
 	}
 
-	if err := image.EnsureSingleRegistry(append(c.additionalTags, c.imageName)...); err != nil {
-		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "all tags must have the same registry as the exported image")
+	if err := image.ValidateDestinationTags(c.useDaemon, append(c.additionalTags, c.imageName)...); err != nil {
+		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "failed to validate image tag(s)")
```
Comment (natalieparellano) [inline cmd/lifecycle/creator.go:81]: Looking at the issue, is the strange phrasing mentioned: `ERROR: failed to all tags must...` going to become `ERROR: failed to failed to validate image tag(s)`? Should the message just start with `validate...`?

Comment (ekcasey) [reply cmd/lifecycle/creator.go:81]: good catch

**Comment by natalieparellano** [comment]
Looks good to me!

**Pull request merged**